content/sensu-go/6.8/operations/deploy-sensu/secure-postgres.md

```diff
@@ -0,0 +1,376 @@
+
+{{< code shell >}}
+export POSTGRES_USERNAME=sensu
+echo '{"CN":"'$POSTGRES_USERNAME'","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -config=ca-config.json -ca=ca.pem -ca-key=ca-key.pem -hostname="" -profile=backend - | cfssljson -bare $POSTGRES_USERNAME
+{{< /code >}}
+
+You should now have the following files in your `/etc/sensu/tls` directory.
+The Sensu backend will use these certificate files to communicate with PostgreSQL:
+
+ filename        | description |
+-----------------|-------------|
+`sensu.pem`      | The certificate that your Sensu backend will use.|
+`sensu-key.pem`  | The private key that your Sensu backend will use. |
+`sensu.csr`      | Certificate signing request for the Sensu backend certificate. Not used. |
+
+Now that you have the required certificates and keys, you can configure Sensu to use certificate authentication with PostgreSQL.
+
+## Configure Sensu to use certificate authentication with PostgreSQL
+
+{{% notice note %}}
+**NOTE**: The Sensu backend uses the libpq library to make connections to PostgreSQL.
+This library [supports a number of environment variables](https://www.postgresql.org/docs/current/libpq-envars.html) that can be injected into the PostgreSQL data source name (DSN) and are loaded at runtime using the system's environment variable file.
+{{% /notice %}}
+
+Working from your Sensu backend, follow these steps to configure Sensu to use certificate authentication with PostgreSQL:
+
+1. Define the environment variables that tell the Sensu backend to use a certificate to authenticate to PostgreSQL:
+
+   {{< language-toggle >}}
+
+{{< code "RHEL/CentOS" >}}
+echo 'PGUSER=sensu
+PGSSLMODE="verify-full"
+PGSSLCERT="/etc/sensu/tls/sensu.pem"
+PGSSLKEY="/etc/sensu/tls/sensu-key.pem"
+PGSSLROOTCERT="/etc/sensu/tls/ca.pem"' | sudo tee /etc/sysconfig/sensu-backend
+{{< /code >}}
+
+{{< code "Ubuntu/Debian" >}}
+echo 'PGUSER=sensu
+PGSSLMODE="verify-full"
+PGSSLCERT="/etc/sensu/tls/sensu.pem"
+PGSSLKEY="/etc/sensu/tls/sensu-key.pem"
+PGSSLROOTCERT="/etc/sensu/tls/ca.pem"' | sudo tee /etc/default/sensu-backend
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+2. Adjust the Sensu datastore connection with sensuctl:
+
+   {{< code shell >}}
+
+echo 'type: PostgresConfig
+api_version: store/v1
+metadata:
+  name: sensu_postgres
+spec:
+  dsn: "postgresql://postgres.example.com:5432/sensu_events"
+  pool_size: 20
+  strict: false' | sudo tee postgresconfig.yml
+
+sensuctl create -f postgresconfig.yml
+
+{{< /code >}}
+
+   {{% notice note %}}
+**NOTE**: Setting `strict: false` in the configuration helps ensure that the Sensu backend will remain active and able to process events even in case of a configuration mistake.
+{{% /notice %}}
+
+3. Confirm that the connection to your PostgreSQL instance is healthy:
+
+   {{< code shell >}}
+curl http://localhost:8080/health
+{{< /code >}}
+
+   The response should be similar to this example, with `true` values for both `Active` and `Healthy`:
+
```

Comment:
Everything works up to this point, but I get an unhealthy PostgreSQL instance:

```
{
  "Alarms": null,
  "ClusterHealth": [
    {
      "MemberID": 13195394291058370000,
      "MemberIDHex": "b71f75320dc06a6c",
      "Name": "default",
      "Err": "",
      "Healthy": true
    }
  ],
  "Header": {
    "cluster_id": 2037210783374497800,
    "member_id": 13195394291058370000,
    "raft_term": 2
  },
  "PostgresHealth": [
    {
      "Name": "sensu_postgres",
      "Active": false,
      "Healthy": false
    }
  ]
}
```

Reply:
Is postgres installed & running?

Reply:
Yes

```
[root@sensu-centos tls]# systemctl status postgresql-14
● postgresql-14.service - PostgreSQL 14 database server
   Loaded: loaded (/usr/lib/systemd/system/postgresql-14.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2022-09-26 19:09:15 UTC; 1h 29min ago
     Docs: https://www.postgresql.org/docs/14/static/
  Process: 4145 ExecStartPre=/usr/pgsql-14/bin/postgresql-14-check-db-dir ${PGDATA} (code=exited, status=0/SUCCESS)
 Main PID: 4150 (postmaster)
   CGroup: /system.slice/postgresql-14.service
           ├─4150 /usr/pgsql-14/bin/postmaster -D /var/lib/pgsql/14/data/
           ├─4152 postgres: logger 
           ├─4154 postgres: checkpointer 
           ├─4155 postgres: background writer 
           ├─4156 postgres: walwriter 
           ├─4157 postgres: autovacuum launcher 
           ├─4158 postgres: stats collector 
           └─4159 postgres: logical replication launcher 

Sep 26 19:09:15 sensu-centos systemd[1]: Starting PostgreSQL 14 database server...
Sep 26 19:09:15 sensu-centos postmaster[4150]: 2022-09-26 19:09:15.600 UTC [4150] LOG:  redirecting log output to logging collector process
Sep 26 19:09:15 sensu-centos postmaster[4150]: 2022-09-26 19:09:15.600 UTC [4150] HINT:  Future log output will appear in directory "log".
Sep 26 19:09:15 sensu-centos systemd[1]: Started PostgreSQL 14 database server.
```